### PR TITLE
kubeadm: update MinimumControlPlaneVersion to v1.11.0

### PR DESCRIPTION
**Pull request by neolit123**
**What this PR does / why we need it**:

Update MinimumControlPlaneVersion to v1.11.0. Also update related
unit tests and test configurations.

based on this PR by @dixudx 
https://github.com/kubernetes/kubernetes/pull/63920
from there adjusting unit tests until nothing fails.

is there anything missing?

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
NONE

**Special notes for your reviewer**:
state of unit tests:
```
$ GOCACHE=off go test ./cmd/kubeam/app/...
?   	k8s.io/kubernetes/cmd/kubeadm/app	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm	0.006s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/fuzzer	0.037s [no tests to run]
?   	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha2	0.023s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha3	0.038s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation	0.087s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/cmd	2.405s
?   	k8s.io/kubernetes/cmd/kubeadm/app/cmd/options	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases	34.704s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/certs	21.058s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade	0.057s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/cmd/util	0.017s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/componentconfigs	0.023s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/constants	0.008s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/discovery	0.023s
?   	k8s.io/kubernetes/cmd/kubeadm/app/discovery/file	[no test files]
?   	k8s.io/kubernetes/cmd/kubeadm/app/discovery/https	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/discovery/token	19.027s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/features	0.003s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/images	0.016s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns	0.018s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy	0.033s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/clusterinfo	0.025s
?   	k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/certs	28.902s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/pkiutil	3.709s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/certs/renewal	2.468s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/controlplane	7.461s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/etcd	0.035s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/kubeconfig	8.008s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet	0.565s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/markmaster	3.537s
?   	k8s.io/kubernetes/cmd/kubeadm/app/phases/patchnode	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/selfhosting	0.037s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade	16.657s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/phases/uploadconfig	0.042s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/preflight	0.040s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util	0.049s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/apiclient	0.028s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/audit	0.013s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/config	0.095s
?   	k8s.io/kubernetes/cmd/kubeadm/app/util/dryrun	[no test files]
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/etcd	0.036s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig	0.028s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/pubkeypin	0.003s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/runtime	0.011s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/staticpod	0.029s
ok  	k8s.io/kubernetes/cmd/kubeadm/app/util/system	0.009s
```

integration tests:
```
$./hack/make-rules/test-kubeadm-cmd.sh

<snip>

+++ [0914 05:20:01] Running tests without code coverage
ok  	k8s.io/kubernetes/cmd/kubeadm/test/cmd	36.049s
make: Leaving directory '/home/lubo-it/go/src/k8s.io/kubernetes'

```

/kind cleanup
/assign @fabriziopandini
/cc @luxas @dixudx @kad
@kubernetes/sig-cluster-lifecycle-pr-reviews 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
i would like to express the _minor concern_ that these kind of updates seem a bit _verbose_.
ideas on what optimization can be done?

File: cmd/kubeadm/app/cmd/phases/certs_test.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ import (
 )
 
 // phaseTestK8sVersion is a fake kubernetes version to use when testing
-const phaseTestK8sVersion = "v1.10.0"
+const phaseTestK8sVersion = "v1.12.0"
```
Comment (xiangpengzhao) [inline cmd/kubeadm/app/cmd/phases/certs_test.go]: Why here jumps to 1.12? Typo?

**Review by xiangpengzhao — COMMENTED** [review]
one nit. otherwise LGTM.

File: cmd/kubeadm/app/cmd/phases/certs_test.go (hunk 1 above)
Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/certs_test.go]: yeah, the tests still pas with it, but probably should either bump to .11 or leave it .10.

**Comment by timothysc** [comment]
@neolit123 Could you include the final version bump that is needed as well and we will hold this PR until the RC's get cut per standard operating procedure. 

/cc @kubernetes/sig-release-admins 
/assign @timothysc 
/hold

**Comment by fabriziopandini** [comment]
Thanks @neolit123!
lgtm

**Review by dixudx — APPROVED** [review]
Great! Thanks @neolit123 
/lgtm

**Review by timothysc — COMMENTED** [review]
Meh, let's merge this now and then do a separate PR on final version bump to make it consistent with history. 

/approve 
/remove hold

**Comment by tpepper** [comment]
Just as an FYI, we're still having some notable test issues on the release, but at this point I do intend to move forward with cutting 1.12's RC tomorrow Sept 18.

**Pull request merged**